client/app/reducers/activity.reducer.js

```diff
@@ -24,13 +23,32 @@ const ActivityReducer = (state = initialState, action) => {
             break;
 
         case ActivityActions.PostActivity.Success:
-            console.log("adding data in reducer *******",action.data)
-            /*return{
-                ...state,
-                activities:
-            };*/
-            duplicateState.activities && duplicateState.activities[0].activities.push(action.data);
-            break;
+            //todo : push activity in its _id:date only  Action.data.date is timestamp and state is in mm/dd/yyyy format
+             if(duplicateState && duplicateState.activities.length>0){
+                 let flag = false;
+                 duplicateState.activities.map((dates)=>{
+                     if(dates._id === action.data.date) {
+                         flag = true;
```

Comment:
No need to use map function here. IndexOf/findIndex function itself achieve the same purpose.